xformers/components/attention/favor.py

```diff
@@ -98,8 +98,7 @@ def __init__(
             "normalize_inputs": self.normalize_inputs,
         }
 
-        self.feature_map_query: FeatureMap = feature_map_constructor(**feature_settings)  # type: ignore
-        self.feature_map_key: FeatureMap = feature_map_constructor(**feature_settings)  # type: ignore
+        self.feature_map: FeatureMap = feature_map_constructor(**feature_settings)  # type: ignore
```

Comment:
oww, that would have been an annoying bug, great catch and I went too fast. I would do a x2 on the setting, because I think that from the outside "iter" means "number of steps we took"

Reply:
I updated the docstring for the attention to make it clear that this counted the number of "forward" calls